### PR TITLE
feat: allow to query app settings by name

### PR DESCRIPTION
**Pull request by spaenleh**
In this PR I add the possibility to query app settings by name. This is done with an optional query parameter. This will allow consumers of the API to get only the list of settings that have a specific name. This will let them type the array as having a specific type linked to that name. This should improve the DX when writing apps.

### CODE REVIEW

File: src/services/item/plugins/app/appData/test/index.test.ts [hunk 1]
```diff
@@ -189,6 +191,22 @@ describe('App Data Tests', () => {
...
+          },
+        });
+        const appDataOfType = appData.filter((d) => d.type === 'other-type');
+        const receivedAppData = await response.json();
+        expect(appDataOfType.length).toBeGreaterThanOrEqual(1);
+        expect(response.statusCode).toEqual(StatusCodes.OK);
+        expectAppData(appDataOfType, receivedAppData);
+        expect(receivedAppData.length).toEqual(appDataOfType.length);
+      });
+
```
Comment (pyphilia) [inline src/services/item/plugins/app/appData/test/index.test.ts:209]: Add a test to check what happens if you don't have matching app data?

File: src/services/item/plugins/app/appSetting/repository.ts [hunk 2]
```diff
@@ -60,13 +60,14 @@ export const AppSettingRepository = AppDataSource.getRepository(AppSetting).exte
...
 
-  getForItem(itemId: string): Promise<AppSetting[]> {
+  getForItem(itemId: string, name?: string): Promise<AppSetting[]> {
     if (!itemId) {
       throw new ItemNotFound(itemId);
     }
     return this.find({
       where: {
         item: { id: itemId },
+        ...(name ? { name } : undefined),
```
Comment (pyphilia) [inline src/services/item/plugins/app/appSetting/repository.ts:70]: I never tried, but does `...undefined` works?

File: src/services/item/plugins/app/appSetting/test/index.test.ts [hunk 3]
```diff
@@ -125,6 +137,20 @@ describe('Apps Settings Tests', () => {
...
+          method: HttpMethod.GET,
+          url: `${APP_ITEMS_PREFIX}/${item.id}/app-settings?name=new-setting`,
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        });
+        expect(response.statusCode).toEqual(StatusCodes.OK);
+        expectAppSettings(response.json(), [appSettings.find((s) => s.name === 'new-setting')]);
+      });
+
```
Comment (pyphilia) [inline src/services/item/plugins/app/appSetting/test/index.test.ts:153]: Add a test with empty result?

**Review by pyphilia — APPROVED** [review]
LGTM, I added some suggestions for the tests, but it looks good! 🎁

File: src/services/item/plugins/app/appSetting/repository.ts (hunk 2 above)
Comment (spaenleh) [reply src/services/item/plugins/app/appSetting/repository.ts:70]: Looks like it does !
